pkg/monitoring/metrics/common/labels/labels_config.go

```diff
@@ -0,0 +1,207 @@
+)
+
+var (
+	DefaultAllowlist  = []string{"*"}
+	DefaultIgnorelist = []string{}
+)
+
+const (
+	ConfigMapName = "kubevirt-vm-labels-config"
+)
+
+type Config interface {
+	ShouldReport(label string) bool
+}
+
+type configImpl struct {
+	mu         sync.RWMutex
+	allowlist  []string
+	ignorelist []string
+	allowAll   bool
+}
+
+// New creates a new labels config instance and, if a client is provided,
+// starts a watcher to keep it updated from the ConfigMap.
+func New(client kubecli.KubevirtClient) (Config, error) {
+	cfg := &configImpl{
+		allowlist:  append([]string{}, DefaultAllowlist...),
+		ignorelist: append([]string{}, DefaultIgnorelist...),
+	}
+	cfg.recomputeAllowAll()
+	if err := cfg.startWatcherWithClient(client); err != nil {
+		return cfg, err
+	}
+
+	return cfg, nil
+}
+
+func (c *configImpl) startWatcherWithClient(client kubecli.KubevirtClient) error {
+	if client == nil {
+		return fmt.Errorf("nil kubevirt client")
+	}
+
+	namespace, err := clientutil.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("failed to determine namespace for watcher: %w", err)
+	}
+
+	lw := cache.NewListWatchFromClient(
+		client.CoreV1().RESTClient(),
+		"configmaps",
+		namespace,
+		fields.OneTermEqualSelector("metadata.name", ConfigMapName),
+	)
+
+	informer := cache.NewSharedIndexInformer(
+		lw,
+		&k8sv1.ConfigMap{},
+		0,
+		cache.Indexers{},
+	)
+
+	c.attachHandlersToInformer(informer)
+	stop := make(chan struct{})
+	go func() {
+		defer close(stop)
+		informer.Run(stop)
+	}()
+	return nil
+}
+
+func (c *configImpl) attachHandlersToInformer(informer cache.SharedIndexInformer) {
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			cm, ok := obj.(*k8sv1.ConfigMap)
+			if !ok {
+				log.Log.Warningf("vm-labels: Add handler received unexpected object type %T", obj)
+				return
+			}
+			c.updateFromConfigMap(cm)
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			cm, ok := newObj.(*k8sv1.ConfigMap)
+			if !ok {
+				log.Log.Warningf("vm-labels: Update handler received unexpected object type %T", newObj)
+				return
+			}
+			c.updateFromConfigMap(cm)
+		},
+		DeleteFunc: func(obj interface{}) {
+			c.resetToDefaults()
+		},
+	})
+}
+
+func (c *configImpl) ShouldReport(label string) bool {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	if len(c.allowlist) == 0 {
+		return false
+	}
+
+	for _, ig := range c.ignorelist {
+		if ig == label {
+			return false
+		}
+	}
+
+	if c.allowAll {
+		return true
+	}
+
+	for _, a := range c.allowlist {
+		if a == label {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (c *configImpl) updateFromConfigMap(configMap *k8sv1.ConfigMap) {
+	if configMap == nil || configMap.Data == nil {
+		c.resetToDefaults()
+		return
+	}
+
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	if d, ok := configMap.Data["allowlist"]; ok {
+		c.allowlist = parseLabels(d)
+	} else {
+		c.allowlist = append([]string{}, DefaultAllowlist...)
+	}
+	c.recomputeAllowAll()
```

Comment:
if you accept my other suggestions
recomputeAllowAll is only called once, i think we can inline